### PR TITLE
Group duplicate logic of WebAppNode and DeploymentSlotNode into WebAppBaseNode

### PR DESCRIPTION
**Pull request by SummerSun**
Create new package `webappbase` and use parent class `WebAppBaseNode` for `WebAppNode` and `DeploymentSlotNode`, and handle the same parts in parent:
- node state and other properties
- the logic of rendering the node by its state
- the logic of processing telemetries (not identical because web app node sends region name)

Move the `WebAppBaseState` to the `webappbase` package.

#2046

### CODE REVIEW

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNodePresenter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public void onStopDeploymentSlot(String subscriptionId, String webAppId, String slotName) throws IOException {
+        //todo
+    }
+
+    public void onRestartDeploymentSlot(String subscriptionId, String webAppId, String slotName) throws IOException {
+        //todo
+    }
+
+    public void onNodeRefresh() {
```
Comment (jdneo) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNodePresenter.java]: onRefreshNode

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNode.java [hunk 2]
```diff
@@ -1,43 +1,55 @@
...
-        properties.put(AppInsightsConstants.SubscriptionId, this.subscriptionId);
-        // todo: track region name
-        return properties;
+    public DeploymentSlotNode(final String slotId, final String webAppId, final DeploymentSlotModule parent,
+                              final String name, final String state, final String os,
+                              final String subscriptionId, final String hostName) {
+        super(slotId, name, LABEL, parent,subscriptionId, hostName, os, state);
+        this.webAppId = webAppId;
+        this.slotName = name;
+        this.presenter =  new DeploymentSlotNodePresenter();
```
Comment (jdneo) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNode.java]: multiple white sapces

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNodePresenter.java (hunk 1 above)
Comment (SummerSun) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNodePresenter.java]: Seems the previous patterns all look like on**NoneVerb**, like the `onModuleRefresh`, `onStartWebApp`.

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotModule.java [hunk 3]
```diff
@@ -35,8 +35,8 @@ protected void refreshItems() throws AzureCmdException {
     }
 
     @Override
-    public void renderDeploymentSlots(@NotNull final List<DeploymentSlot> slots) {
-        slots.forEach(slot -> addChildNode(new DeploymentSlotNode(
-            this, slot.name(), slot.state(), slot.operatingSystem().toString(), this.subscriptionId)));
+    public void renderDeploymentSlots(final String webAppId, @NotNull final List<DeploymentSlot> slots) {
```
Comment (Eskibear) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotModule.java]: no need to pass webappid

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNodePresenter.java (hunk 1 above)
Comment (Eskibear) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNodePresenter.java]: `onRefreshNode` would be better, see `onStartDeploymentSlot`, `onSwapWithProduction` ..etc in this file

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotModule.java (hunk 3 above)
Comment (SummerSun) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotModule.java]: yes, will get it from `slot.parent().id()`

**Pull request merged**